.circleci/config.yml

```diff
@@ -334,20 +334,36 @@ jobs:
 
   # We need a full clone here to be able to compare to master.
   spellcheck:
-    <<: *defaults
+    docker:
+      - image: stellargroup/build_env:14
+    resource_class: large  # Increased memory allocation
     steps:
       - checkout:
           path: /hpx/source
+          fetch-depth: 0  # Ensure full history for proper diff
       - run:
           name: Running codespell
           command: |
               if [[ -n "$CIRCLE_PULL_REQUEST" ]]; then
                   cd /hpx/source
                   codespell --version
-                  codespell --ignore-words tools/.codespell_whitelist --skip='*.h5,*.png' $(git diff --name-only origin/master...) > /tmp/spelling_suggestions.txt
+                  
+                  echo "Fetching changed files from origin/master..."
+                  git fetch origin master --depth=1
+                  
+                  echo "Files being checked:"
+                  git diff --name-only origin/master... | tee /tmp/file_list.txt
+
+                  echo "Limiting to 50 files for memory efficiency..."
+                  head -n 50 /tmp/file_list.txt > /tmp/limited_file_list.txt
+
+                  set -x  # Debugging mode enabled
+                  codespell --ignore-words tools/.codespell_whitelist --skip='*.h5,*.png' -f /tmp/limited_file_list.txt > /tmp/spelling_suggestions.txt
+                  set +x  # Disable debugging
+
```

Comment:
Can you change this to go through all changed files 50 files at a time (not just processing the 50 first files in the list)?

Reply:
Wouldn't it still work only on the first 50 files in the list? `FILES` is initialized from whatever `head -n 50 ...` gives you.